test/stmt/then_stmt_disabled.swift

```diff
@@ -0,0 +1,11 @@
+// RUN: %target-typecheck-verify-swift
+
+// Then statements are disabled by default
+let x = if .random() {
+  print("hello")
+  then 0
+  // expected-error@-1 {{cannot find 'then' in scope}}
```

Comment:
Should this be diagnosed as a disabled feature instead if `then` here otherwise would be valid?

Reply:
This is a source breaking change, so we can't generally diagnose it as a disabled feature, we could apply a heuristic to generate that diagnostic, but I'm not convinced it's worth it.

Reply:
Maybe it's not worth it if we are going to enable this by default soon, but it would be great for user experience to hint that all they need is to enable a flag in situations when it's syntactically unambiguous that they meant `then` statement.